src/vvwe.rs

```diff
@@ -254,10 +238,11 @@ mod test {
         let del = CausalMessage {
             time: 1,
             local_id: 1,
-            op: Op::Delete(1, 1),
+            op: Op::Delete(1, 0),
         };
-        assert_eq!(barrier.ingest(ins.clone()), Some(ins));
-        assert_eq!(barrier.ingest(del.clone()), Some(del));
+        assert_eq!(barrier.ingest(ins.clone()), vec![ins]);
+        assert_eq!(barrier.ingest(del.clone()), vec![del]);
+        assert!(barrier.is_empty());
```

Comment:
I realized that one of the tests was passing asserts because it wasn't actually ever emptying out the buffer! Now I check that the buffer is properly emptied at the end of the asserts.